src/commands/scan/handle-create-new-scan.mts

```diff
@@ -106,8 +111,29 @@ export async function handleCreateNewScan({
     return
   }
 
+  let scanPaths: string[] = packagePaths
+
+  // If reachability is enabled, perform reachability analysis
+  if (reach) {
+    const reachResult = await performReachabilityAnalysis({
+      packagePaths,
+      orgSlug,
+      cwd,
+      repoName,
+      branchName,
+      outputKind,
+      interactive,
+    })
+
+    if (!reachResult.ok || !reachResult.scanPaths) {
```

Comment:
I guess you're doing this because of TS.
If the function returns a CResult<{scanPaths}> then this extra check wouldn't be necessary. I'm expecting this to just do `if (reachResult.ok)`